--- Makefile
+++ Makefile
@@ -56,7 +56,7 @@
   CFLAGS += -g -DDEBUG
 else
   STRIP = strip
-  CFLAGS += -O
+  CFLAGS += -Wall
 endif
 
 # default prefix
@@ -131,7 +131,7 @@
 
 CFLAGS += -I $(SUBDIR)
 CPPFLAGS = $(CFLAGS)
-LDFLAGS = -lm $(LLSCF)
+LDFLAGS += -lm $(LLSCF)
 
 ifneq ($(BUILD_OS),FreeBSD)
 LDFLAGS += -ldl
--- XF86Config-parser/Makefile
+++ XF86Config-parser/Makefile
@@ -20,7 +20,6 @@
 
 OBJS = $(SRC:.c=.o)
 
-CFLAGS = -Wall -g
 
 ifdef NV_CFLAGS
   CFLAGS += $(NV_CFLAGS)
