--- /var/tmp/portage/libquicktime-0.9.9/work/libquicktime-0.9.9/plugins/x264/x264.c.old	2006-06-25 15:10:30.000000000 +0200
+++ /var/tmp/portage/libquicktime-0.9.9/work/libquicktime-0.9.9/plugins/x264/x264.c	2006-08-12 13:12:31.000000000 +0200
@@ -89,7 +89,7 @@
   fprintf(stderr, "    i_qp_min:           %d\n", params->rc.i_qp_min);
   fprintf(stderr, "    i_qp_max:           %d\n", params->rc.i_qp_max);
   fprintf(stderr, "    i_qp_step:          %d\n", params->rc.i_qp_step);
-  fprintf(stderr, "    b_cbr:              %d\n", params->rc.b_cbr);
+//  fprintf(stderr, "    b_cbr:              %d\n", params->rc.b_cbr);
   fprintf(stderr, "    i_bitrate:          %d\n", params->rc.i_bitrate);
   fprintf(stderr, "    i_rf_constant:      %d\n", params->rc.i_rf_constant);
   fprintf(stderr, "    f_rate_tolerance:   %f\n", params->rc.f_rate_tolerance);
@@ -641,12 +641,12 @@
     {
     if(*(int*)(value))
       {
-      codec->params.rc.b_cbr = 1;
+//      codec->params.rc.b_cbr = 1;
       codec->params.rc.i_bitrate = *(int*)(value);
       }
     else
       {
-      codec->params.rc.b_cbr = 0;
+//      codec->params.rc.b_cbr = 0;
       codec->params.rc.i_bitrate = 0;
       }
     found = 1;

