--- src/buffer.orig	2005-11-12 17:25:50.000000000 +0800
+++ src/buffer.c	2005-11-12 17:27:07.000000000 +0800
@@ -314,8 +314,11 @@
   read_more:
   /* Make sure we have space available to read(2) data into */
   remain = mcm_buf_remain(ctxt, buf);
-  if (remain == 0 && !mcm_buf_realloc(ctxt, buf, mcm_buf_size(ctxt, buf) * 2)) /* Double our size */
-    return bytes_read;
+  if (remain == 0) {
+	  if (!mcm_buf_realloc(ctxt, buf, mcm_buf_size(ctxt, buf) * 2)) /* Double our size */
+		  return bytes_read;
+	  remain = mcm_buf_remain(ctxt, buf);
+  }
 
   rb = read(fd, mcm_buf_tail(ctxt, buf), remain);
 #ifdef DEBUG_MC_PROTO
